src/verify_cert.rs

```diff
@@ -1102,6 +1068,34 @@ mod tests {
         .map(|_| ())
     }
 
+    fn build_linear_chain(
```

Comment:
Nit: I think a further improvement here would be to instead yield a `struct IntermediateChain { last: rcgen::CertifiedKey, chain: Vec<CertificateDer<'static>> }` instead? At any point, it seems like we only need to hold one `CertifiedKey`.

Reply:
Implemented :+1: